### PR TITLE
OSSM-8318 Explanation of Kiali CR in TP1 doc

### PR DESCRIPTION
**Pull request by gwynnemonahan**
**OSSM 3.0 TP1**

[OSSM-8318](https://issues.redhat.com//browse/OSSM-8318) Explanation of Kiali CR in TP1 doc

**Merge to**: https://github.com/openshift/openshift-docs/tree/service-mesh-docs-main

**Cherry pick**: to https://github.com/openshift/openshift-docs/tree/service-mesh-docs-3.0.0tp1

This PR is part of the standalone doc set for the OpenShift Service Mesh project. Kathryn is aware that this content applies for a product that is part of a Technology Preview release. The project is seeking feedback from early adopters.

Version(s):

Technology Preview.

OSSM 3.0 is moving to stand alone format is will not be cherry-picked back to OCP core branches.

Issue:
https://issues.redhat.com/browse/OSSM-8318 

Link to docs preview:

Kiali preview: https://84255--ocpdocs-pr.netlify.app/openshift-service-mesh/latest/observability/kiali/ossm-kiali-assembly.html#ossm-config-otel-kiali_ossm-kiali-assembly

Tracing preview, go to Step 6 to see NOTE added: https://84255--ocpdocs-pr.netlify.app/openshift-service-mesh/latest/observability/traces/ossm-distr-tracing-assembly#ossm-config-otel_ossm-traces-assembly

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/ossm-config-otel-kiali.adoc [hunk 1]
```diff
@@ -23,22 +23,18 @@ After {KialiProduct} is integrated with {DTProductName}, you can view distribute
...
-      url: http://grafana-istio-system.apps-crc.testing/
     tracing:
-      enabled: true
+      enabled: true #<1>
       provider: tempo
       use_grpc: false
       in_cluster_url: http://tempo-sample-query-frontend.tempo:3200
-      url: https://tempo-sample-query-frontend-tempo.apps-crc.testing
-      tempo_config: # <3>
+      url: https://tempo-sample-query-frontend-tempo.apps-crc.testing #<2>
```
Comment (josunect) [inline modules/ossm-config-otel-kiali.adoc:31]: This won't be needed

File: modules/ossm-config-otel-kiali.adoc [hunk 2]
```diff
@@ -23,22 +23,18 @@ After {KialiProduct} is integrated with {DTProductName}, you can view distribute
...
+      tempo_config:
```
Comment (josunect) [inline modules/ossm-config-otel-kiali.adoc:32]: All the tempo config won't be needed, it is used to create the Tempo datasource for the Grafana url

File: modules/ossm-config-otel-kiali.adoc [hunk 3]
```diff
@@ -23,22 +23,18 @@ After {KialiProduct} is integrated with {DTProductName}, you can view distribute
...
         org_id: "1"
-        datasource_uid: a8d2ef1c-d31c-4de5-a90b-e7bc5252cd00
+        datasource_uid: a8d2ef1c-d31c-4de5-a90b-e7bc5252cd00 #<3>
 ----
-<1> Required so that you can see the *View in Tracing* link from the *Traces* tab.
-<2> If the Tempo resource is deployed with multiple tenants, then the `in_cluster_url` config should point to the gateway with the tenant name as part of the URL. For example: http://tempo-resource-gateway.tempo:8080/api/traces/v1/{tenantName}/tempo
+<1> Enable tracing.
+<2> The OpenShift route for Jaeger UI must be created in the Tempo namespace. You can either manually create it for `tempo-sample-query-frontend` service, or update the `Tempo` custom resource with `.spec.template.queryFrontend.jaegerQuery.ingress.type: route`.
```
Comment (josunect) [inline modules/ossm-config-otel-kiali.adoc:37]: This <2> won't be needed

File: modules/ossm-config-otel-kiali.adoc (hunk 1 above)
Comment (mkralik3) [reply modules/ossm-config-otel-kiali.adoc:31]: Isn't that URL ( in `.tracing.url`) used in `Distributed Tracing` link in Kiali UI ? ( and when it is not set, the `Distributed Tracing` link is missing in Kiali UI ? ) 
image

( or does it work that way only for KIali 1.73? and for 1.89> it works differently? [I have not tried ossm3 + kiali 1.89 + tempo yet, I am going to do today] )

Comment (josunect) [reply modules/ossm-config-otel-kiali.adoc:31]: Ah right @mkralik3 , I've missed this. Let me update the comments

**Review by josunect — APPROVED** [review]
LGTM!

**Comment by mkralik3** [comment]
LGTM

**Comment by gwynnemonahan** [comment]
Rebased 11/01/2024

File: modules/ossm-config-otel.adoc [hunk 4]
```diff
@@ -146,5 +146,10 @@ $ oc exec -it -n bookinfo deployments/productpage-v1 -c istio-proxy -- curl loca
 +
 [source,terminal]
 ----
-$ kubectl get routes -n tempo tempo-sample-query-frontend
-----
\ No newline at end of file
+$ oc get routes -n tempo tempo-sample-query-frontend
```
Comment (adellape) [inline modules/ossm-config-otel.adoc:149]: Technically out of scope for the PR, but if you're in here, I notice a handful of other command blocks in this procedure module that are missing their `$` prompt, and an instance of `<revision-name>` that should be `<revision_name>`.

File: modules/ossm-config-otel-kiali.adoc [hunk 5]
```diff
@@ -23,31 +23,23 @@ After {KialiProduct} is integrated with {DTProductName}, you can view distribute
...
 
 . Run the following command to apply the configuration:
 +
 [source, terminal]
 ----
-oc patch -n istio-system kiali kiali --type merge -p "$(cat kiali_cr.yaml)"
+$ oc patch -n istio-system kiali kiali --type merge -p "$(cat kiali_cr.yaml)"
 ----
 +
 Output:
```
Comment (adellape) [inline modules/ossm-config-otel-kiali.adoc:45]:
```suggestion
.Example output
```

File: modules/ossm-config-otel.adoc (hunk 4 above)
Comment (gwynnemonahan) [reply modules/ossm-config-otel.adoc:149]: I noticed a few were missing while updating, no idea how it made it through peer review and merge review the first time! :laughing: 

I was adding them in as I noticed them. Seems I missed some. Thanks!

Comment (gwynnemonahan) [reply modules/ossm-config-otel.adoc:149]: <revision-name> is correct, see

oc label namespace <namespace-name> istio.io/rev=<revision-name> here: https://83665--ocpdocs-pr.netlify.app/openshift-service-mesh/latest/update/ossm-updating-openshift-service-mesh

File: modules/ossm-config-otel.adoc [hunk 6]
```diff
@@ -146,5 +146,10 @@ $ oc exec -it -n bookinfo deployments/productpage-v1 -c istio-proxy -- curl loca
...
+----
++
+[NOTE]
+====
+The OpenShift route for Jaeger UI must be created in the Tempo namespace. You can either manually create it for `tempo-sample-query-frontend` service, or update the `Tempo` custom resource with `.spec.template.queryFrontend.jaegerQuery.ingress.type: route`.
```
Comment (jeana-redhat) [inline modules/ossm-config-otel.adoc:154]: Should "OpenShift" be an attribute?

Comment (jeana-redhat) [inline modules/ossm-config-otel.adoc:154]:
```suggestion
The OpenShift route for Jaeger UI must be created in the Tempo namespace. You can either manually create it for the `tempo-sample-query-frontend` service, or update the `Tempo` custom resource with `.spec.template.queryFrontend.jaegerQuery.ingress.type: route`.
```

File: modules/ossm-config-otel-kiali.adoc [hunk 7]
```diff
@@ -23,34 +23,26 @@ After {KialiProduct} is integrated with {DTProductName}, you can view distribute
...
-      tempo_config: # <3>
-        org_id: "1"
-        datasource_uid: a8d2ef1c-d31c-4de5-a90b-e7bc5252cd00
+      url: https://tempo-sample-query-frontend-tempo.apps-crc.testing #<2>
 ----
-<1> Required so that you can see the *View in Tracing* link from the *Traces* tab.
-<2> If the Tempo resource is deployed with multiple tenants, then the `in_cluster_url` config must point to the gateway with the tenant name as part of the URL. For example: http://tempo-resource-gateway.tempo:8080/api/traces/v1/{tenantName}/tempo
-<3> Required so that you can see the *View in Tracing*  link from the *Traces* tab and redirect to the proper Tempo `datasource`.
+<1> Enable tracing.
+<2> The OpenShift route for Jaeger UI must be created in the Tempo namespace. You can either manually create it for `tempo-sample-query-frontend` service, or update the `Tempo` custom resource with `.spec.template.queryFrontend.jaegerQuery.ingress.type: route`.
```
Comment (jeana-redhat) [inline modules/ossm-config-otel-kiali.adoc:34]: Attribute for OpenShift? and:
```suggestion
<2> The OpenShift route for Jaeger UI must be created in the Tempo namespace. You can either manually create it for the `tempo-sample-query-frontend` service, or update the `Tempo` custom resource with `.spec.template.queryFrontend.jaegerQuery.ingress.type: route`.
```

**Review by jeana-redhat — COMMENTED** [review]
Two tiny nits!

/remove-label merge-review-in-progress
/remove-label merge-review-needed

**Pull request merged**